File: 프로그래머스/unrated/181930. 주사위 게임 2/주사위 게임 2.java
class Solution {
    public int solution(int a, int b, int c) {
        int ans = a + b + c;
        if (a != b && b != c && c != a)
            return ans;
        ans *= (a * a + b * b + c * c);
        if (a != b || b != c)
            return ans;
        else
            return ans * (a * a * a + b * b * b + c * c * c);
    }
}

File: 프로그래머스/Lv.0/181930. 주사위 게임 2/주사위 게임 2.js
function solution(a, b, c) {
    var p = a === b;
    var q = b === c;
    var r = c === a;
    var answer = (a + b + c);
    if (!p && !q && !r)
        return answer;
    answer *= (a ** 2 + b ** 2 + c ** 2);
    if (!p || !q || !r)
        return answer;
    answer *= (a ** 3 + b ** 3 + c ** 3);
    return answer;
}